backend/src/test/java/com/github/wekaito/backend/ProfileControllerTest.java

```diff
@@ -0,0 +1,56 @@
+package com.github.wekaito.backend;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+class ProfileControllerTest {
+
+    @Autowired
+    MockMvc mockMvc;
+
+    @Autowired
+    ObjectMapper objectMapper;
+
+    @Test
+    void expectArrayOfCards_whenGetCards() throws Exception {
```

Comment:
Du solltest wenn du in einem Test eine externe API aufrufst diese mocken. 
Guck dir dazu doch den mockwebserver an...
https://github.com/bartfastiel/integrationtest-externe-api/blob/main/src/test/java/com/example/integrationtestexterneapi/RickAndMortyIntegrationTest.java